python/tests/unittests/test_NoiseModel.py

```diff
@@ -41,11 +40,166 @@ def test_Depol():
     assert (len(counts) == 1)
     assert ('1' in counts)
 
+    cudaq.unset_noise()
```

Comment:
I wonder if these `unset_noise()` calls are necessary since the noise model was set locally in the `sample()` call which should have reset the noise model on return.